src/snowflake/snowpark/modin/plugin/_internal/cut_utils.py

```diff
@@ -188,87 +184,37 @@ def compute_bin_indices(
     # consequences when it comes to building lazy graphs, as both cut and qcut are materializing operations.
 
     cuts_frame = cuts_frame.ensure_row_position_column()
-    value_frame = values_frame.ensure_row_position_column()
-
-    (
-        bucket_data_identifier,
-        bucket_row_position_identifier,
-        value_data_identifier,
-        value_row_position_identifier,
-    ) = value_frame.ordered_dataframe.generate_snowflake_quoted_identifiers(
-        pandas_labels=["b_data", "b_row_pos", "v_data", "v_row_pos"]
-    )
-
-    value_index_identifiers = value_frame.index_column_snowflake_quoted_identifiers
-
-    bucket_snowpark_frame = (
-        cuts_frame.ordered_dataframe.to_projected_snowpark_dataframe(True, True, True)
-    )
-    value_snowpark_frame = (
-        value_frame.ordered_dataframe.to_projected_snowpark_dataframe(True, True, True)
-    )
-
-    # relabel to new identifiers to reference within range join below.
-    bucket_snowpark_frame = bucket_snowpark_frame.select(
-        col(cuts_frame.data_column_snowflake_quoted_identifiers[0]).as_(
-            bucket_data_identifier
-        ),
-        col(cuts_frame.row_position_snowflake_quoted_identifier).as_(
-            bucket_row_position_identifier
-        ),
-    )
-
-    value_snowpark_frame = value_snowpark_frame.select(
-        *tuple(value_index_identifiers),
-        col(value_frame.data_column_snowflake_quoted_identifiers[0]).as_(
-            value_data_identifier
-        ),
-        col(value_frame.row_position_snowflake_quoted_identifier).as_(
-            value_row_position_identifier
-        ),
-    )
-
-    # Perform a left join. The idea is to find all values which fall into an interval
-    # defined by the cuts/bins in the bucket frame. The closest can be then identified using the
-    # row position. An alternative to this
-    # was to use an ASOF join with a proper matching condition.
-
+    # perform asof join to find the closet to the cut frame data.
     if right:
-        ans = value_snowpark_frame.join(
-            bucket_snowpark_frame,
-            value_snowpark_frame[value_data_identifier]
-            <= bucket_snowpark_frame[bucket_data_identifier],
-            how="left",
-            lsuffix="_L",
-            rsuffix="_R",
+        asof_result = join(
```

Comment:
nit: this can be simplified to 

```
 asof_result = join(
            values_frame,
            cuts_frame,
            how="asof",
            left_on=[],
            right_on=[],
            left_match_col=values_frame.data_column_snowflake_quoted_identifiers[0],
            right_match_col=cuts_frame.data_column_snowflake_quoted_identifiers[0],
            match_comparator=MatchComparator.LESS_THAN_OR_EQUAL_TO if right else MatchComparator.GREATER_THAN_OR_EQUAL_TO
            sort=False,
        )
```

Reply:
good point, updated